client/my-sites/domains/domain-management/settings/cards/domain-redirect-card.tsx

```diff
@@ -103,10 +102,28 @@ export default function DomainRedirectCard( {
 
 	const handleChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
 		setTargetUrl( withoutHttp( event.target.value ) );
-		if (
-			event.target.value.length > 0 &&
-			! CAPTURE_URL_RGX.test( protocol + '://' + event.target.value )
-		) {
+
+		if ( event.target.value.length === 0 ) {
+			setIsValidUrl( true );
+			return;
+		}
+
+		try {
+			// Use an invalid origin as the URL(,base), if the URL is valid it will be
+			// replaced with the correct origin.
+			const url = new URL( protocol + '://' + event.target.value, 'https://_domain_.invalid' );
+			if ( url.origin === 'https://_domain_.invalid' ) {
```

Comment:
Since it was allowing us adding spaces on the URL I created this "soft" regex that seems to fix our issues: 

<img width="919" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1044309/5baa9c1a-7ced-4610-bad3-b1cf44792b1e">